prow/spyglass/viewers/buildlog/buildlog_viewer.go

```diff
@@ -158,68 +197,21 @@ func ViewHandler(artifacts []viewers.Artifact, raw string) string {
 
 // nextViewData constructs the new log artifact view and needed requests from the current
 // view data and artifact
-func nextViewData(artifact viewers.Artifact, currentViewData LogViewData) (LogArtifactView, LogViewData) {
-	var logLines []string
-	var err error
+func nextViewData(logIndex int, artifact viewers.Artifact, currentViewData LogViewData) (LogArtifactView, LogViewData) {
 	newLogViewData := LogViewData{
 		Operation: "",
 	}
 	newArtifactView := LogArtifactView{
 		ArtifactName: artifact.JobPath(),
 		ArtifactLink: artifact.CanonicalLink(),
 		ViewName:     name,
+		Index:        logIndex,
 	}
-	switch currentViewData.Operation {
-	case "all":
-		logLines = logLinesAll(artifact)
-		newArtifactView.ViewMethodDescription = fmt.Sprintf("viewing all lines")
-		newArtifactView.ViewAll = true
-	case "more":
-		if currentViewData.UseBytes {
-			nBytes := int64(currentViewData.HeadBytes + byteChunkSize)
-			artifactSize, err := artifact.Size()
-			if err != nil {
-				logrus.WithFields(logrus.Fields{"artifact": artifact.JobPath()}).WithError(err).Error("Error getting artifact size.")
-			}
-			if nBytes >= artifactSize {
-				logLines = logLinesAll(artifact)
-				newArtifactView.ViewMethodDescription = fmt.Sprintf("viewing all lines")
-				newArtifactView.ViewAll = true
-			} else {
-				newArtifactView.ViewMethodDescription = fmt.Sprintf("viewing first %d bytes", nBytes)
-				newLogViewData.HeadBytes = nBytes
-				newLogViewData.UseBytes = true
-				logLines, err = logLinesHead(artifact, nBytes)
-				if err != nil {
-					logrus.WithError(err).Error("Reading log lines failed.")
-					newArtifactView.ViewMethodDescription = fmt.Sprintf("failed to read log file - is it compressed?")
-					logLines = []string{}
-				}
-			}
-		} else {
-			nLines := currentViewData.TailLines + lineChunkSize
-			logLines, err = viewers.LastNLines(artifact, nLines)
-			if err != nil {
-				nBytes := int64(byteChunkSize)
-				newLogViewData.UseBytes = true
-				newLogViewData.HeadBytes = nBytes
-				newArtifactView.ViewMethodDescription = fmt.Sprintf("viewing first %d bytes", nBytes)
-				logLines, err = logLinesHead(artifact, nBytes)
-				if err != nil {
-					logrus.WithError(err).Error("Reading head of log lines failed.")
-					newArtifactView.ViewMethodDescription = fmt.Sprintf("failed to read log file - is it compressed?")
-					logLines = []string{}
-				}
-			} else {
-				newLogViewData.TailLines = nLines
-				newArtifactView.ViewMethodDescription = fmt.Sprintf("viewing last %d lines", nLines)
-			}
-		}
-	case "":
-	default:
-	}
```

Comment:
Oh thank the glorious Gopher this block of code is gone, not my proudest hack by a long shot :facepalm: